In [2]:
aws_sales=pd.read_excel("AWS_Sales.xlsx",
    sheet_name="aws_sales")

customers = pd.read_excel(
    "AWS_Sales.xlsx",
    sheet_name="customers"
)
calendar = pd.read_excel(
    "AWS_Sales.xlsx",
    sheet_name="calendar"
)
territories = pd.read_excel(
    "AWS_Sales.xlsx",
    sheet_name="territories"
)
products_cat = pd.read_excel(
    "AWS_Sales.xlsx",
    sheet_name="products_cat"
)
products_subcat = pd.read_excel(
    "AWS_Sales.xlsx",
    sheet_name="products_subcat"
)
products = pd.read_excel(
    "AWS_Sales.xlsx",
    sheet_name="products"
)
returns = pd.read_excel(
    "AWS_Sales.xlsx",
    sheet_name="returns"
)


C:\Users\shukl\anaconda3\Lib\site-packages\pandas\core\computation\expressions.py:23: UserWarning: Pandas requires version '2.10.2' or newer of 'numexpr' (version '2.8.4' currently installed).
  from pandas.core.computation.check import NUMEXPR_INSTALLED
C:\Users\shukl\anaconda3\Lib\site-packages\pandas\core\arrays\masked.py:56: UserWarning: Pandas requires version '1.4.2' or newer of 'bottleneck' (version '1.3.5' currently installed).
  from pandas.core import (


<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [3]:
product_master = (
    products
    .merge(
        products_subcat,
        on='ProductSubcategoryKey',
        how='left'
    )
    .merge(
        products_cat,
        on='ProductCategoryKey',
        how='left'
    )
)

In [4]:
product_master

,ProductKey,ProductSubcategoryKey,ProductSKU,ProductName,ModelName,ProductDescription,ProductColor,ProductSize,ProductStyle,ProductCost,ProductPrice,Discounted_Price,Profit_Before_Discount,Profit_After_Discount,SubcategoryName,ProductCategoryKey,CategoryName
0,214,31,HL-U509-R,"Sport-100 Helmet, Red",Sport-100,"Universal fit, well-vented, lightweight , snap...",Red,0,0,13.1,35.0,31.50,21.9,18.40,Helmets,4,Accessories
1,215,31,HL-U509,"Sport-100 Helmet, Black",Sport-100,"Universal fit, well-vented, lightweight , snap...",Black,0,0,12.0,33.6,30.24,21.6,18.24,Helmets,4,Accessories
2,220,31,HL-U509-B,"Sport-100 Helmet, Blue",Sport-100,"Universal fit, well-vented, lightweight , snap...",Blue,0,0,12.0,33.6,30.24,21.6,18.24,Helmets,4,Accessories
3,223,19,CA-1098,AWC Logo Cap,Cycling Cap,Traditional style with a flip-up brim; one-siz...,Multi,0,U,5.7,8.6,7.74,2.9,2.04,Caps,3,Clothing
4,391,10,FK-1639,LL Fork,LL Fork,Stout design absorbs shock and offers more pre...,NaN,0,0,65.8,148.2,133.38,82.4,67.58,Forks,2,Components
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
288,599,1,BK-M18B-48,"Mountain-500 Black, 48",Mountain-500,"Suitable for any type of riding, on or off-roa...",Black,48,U,294.6,540.0,486.00,245.4,191.40,Mountain Bikes,1,Bikes
289,600,1,BK-M18B-52,"Mountain-500 Black, 52",Mountain-500,"Suitable for any type of riding, on or off-roa...",Black,52,U,294.6,540.0,486.00,245.4,191.40,Mountain Bikes,1,Bikes
290,604,2,BK-R19B-44,"Road-750 Black, 44",Road-750,Entry level adult bike; offers a comfortable r...,Black,44,U,343.6,540.0,486.00,196.4,142.40,Road Bikes,1,Bikes
291,605,2,BK-R19B-48,"Road-750 Black, 48",Road-750,Entry level adult bike; offers a comfortable r...,Black,48,U,343.6,540.0,486.00,196.4,142.40,Road Bikes,1,Bikes


In [5]:
sales_master=(
    aws_sales.merge(customers,on='CustomerKey',how='left')
    .merge(product_master,on='ProductKey',how='left')
    .merge(territories,left_on='TerritoryKey',right_on='SalesTerritoryKey',how='left')
)

In [7]:
sales_master.shape

(56046, 53)

In [10]:
sales_master.columns.tolist()

['Source.Name',
 'OrderDate',
 'StockDate',
 'OrderNumber',
 'ProductKey',
 'CustomerKey',
 'TerritoryKey',
 'OrderLineItem',
 'OrderQuantity',
 'QuanityType',
 'RelatedPrice',
 'RelatedDayName',
 'Sales',
 'Prefix',
 'FirstName',
 'LastName',
 'FullName',
 'BirthDate',
 'MaritalStatus',
 'Gender',
 'EmailAddress',
 'AnnualIncome',
 'TotalChildren',
 'EducationLevel',
 'Occupation',
 'HomeOwner',
 'UserName',
 'DomainName',
 'BirthYear_CC',
 'CurrentAge',
 'BirthMonth',
 'Parent',
 'Full_Name_CC',
 'ProductSubcategoryKey',
 'ProductSKU',
 'ProductName',
 'ModelName',
 'ProductDescription',
 'ProductColor',
 'ProductSize',
 'ProductStyle',
 'ProductCost',
 'ProductPrice',
 'Discounted_Price',
 'Profit_Before_Discount',
 'Profit_After_Discount',
 'SubcategoryName',
 'ProductCategoryKey',
 'CategoryName',
 'SalesTerritoryKey',
 'Region',
 'Country',
 'Continent']

In [8]:
aws_sales.shape

(56046, 13)

In [9]:
# Check null values
sales_master.isnull().sum().sort_values(ascending=False )

ProductColor              26878
Prefix                      380
Gender                      380
ProductStyle                  0
BirthMonth                    0
Parent                        0
Full_Name_CC                  0
ProductSubcategoryKey         0
ProductSKU                    0
ProductName                   0
ModelName                     0
ProductDescription            0
ProductSize                   0
Source.Name                   0
BirthYear_CC                  0
ProductCost                   0
ProductPrice                  0
Discounted_Price              0
Profit_Before_Discount        0
Profit_After_Discount         0
SubcategoryName               0
ProductCategoryKey            0
CategoryName                  0
SalesTerritoryKey             0
Region                        0
Country                       0
CurrentAge                    0
UserName                      0
DomainName                    0
Sales                         0
StockDate                     0
OrderNum

1.ProductColor contains null values for approximately 48% of records, likely because color is not applicable to all product types.
2.Less than 1% of customer records contain missing gender and prefix information.